front50-web/src/main/groovy/com/netflix/spinnaker/front50/graphql/datafetcher/PipelinesDataFetcher.java

```diff
@@ -42,18 +39,44 @@ public PipelinesDataFetcher(PipelineDAO pipelineDAO) {
   public Collection<Pipeline> get(DataFetchingEnvironment environment) {
     Collection<Pipeline> pipelines = pipelineDAO.all();
 
-    Object havingTriggerType = environment.getArgument("havingTriggerType");
-    if (havingTriggerType != null) {
-      return pipelines.stream()
-        .filter(pipeline -> {
-          List<Map<String, Object>> triggers = (List<Map<String, Object>>) pipeline.getOrDefault("triggers", Collections.EMPTY_LIST);
-          return triggers.stream().anyMatch(trigger -> havingTriggerType.equals(trigger.get("type")));
-        })
-        .collect(Collectors.toList());
+    // TODO rz - mapping stuff like `extra` and other nested subtypes correctly.
+
+    return pipelines.stream()
+      .filter(p ->
+        new HavingTriggerTypePredicate(environment)
+          .and(new DisabledPredicate(environment))
+          .test(p)
+      )
+      .collect(Collectors.toList());
+  }
+
+  static class HavingTriggerTypePredicate extends DataFetchingEnvironmentPredicate<Pipeline> {
+    HavingTriggerTypePredicate(DataFetchingEnvironment environment) {
+      super(environment);
     }
 
-    // TODO rz - mapping stuff like `extra` and other nested subtypes correctly.
+    @Override
+    public boolean test(Pipeline pipeline) {
+      String havingTriggerType = environment.getArgument("havingTriggerType");
+      if (havingTriggerType == null) {
+        return true;
+      }
+      return pipeline.getTriggers().stream().anyMatch(t -> havingTriggerType.equals(t.getType()));
+    }
+  }
 
-    return pipelines;
+  static class DisabledPredicate extends DataFetchingEnvironmentPredicate<Pipeline> {
+    DisabledPredicate(DataFetchingEnvironment environment) {
+      super(environment);
+    }
+
+    @Override
+    public boolean test(Pipeline pipeline) {
+      if (!environment.containsArgument("disabled")) {
+        return true;
+      }
+      boolean disabled = environment.getArgument("disabled");
+      return (Boolean) pipeline.getOrDefault("disabled", false) == disabled;
```

Comment:
I don't think the cast to `(Boolean)` is needed

Reply:
It is needed. `Pipeline` is a `HashMap<String, Object>`, so even if the default value is a `boolean`, the returned type of `getOrDefault` will still be `Object`.

Reply:
oh boy, I misinterpreted the operator precedence here. I thought the case applied to the whole `==` expression, not just the left hand side.